### PR TITLE
Update the object initializer BBE

### LINKED ISSUE #14661: Update the BBE for objects to reflect changes to the __init() return signature
Need to do $subject

### PR DESCRIPTION
**Pull request by pubudu91**
Fixes #14661

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [x] Ballerina By Examples

### CODE REVIEW

File: examples/object-initializer/object_initializer.description [hunk 1]
```diff
@@ -1 +1,4 @@
-// In Ballerina, objects can have a constructor that can be used to initialize the object.
+// In Ballerina, objects have a method named `__init()` which is used to initialize the object when creating a
```
Comment (MaryamZi) [inline examples/object-initializer/object_initializer.description]:
```suggestion
// In Ballerina, objects can have a method named `__init()` which is used to initialize the object when creating a
```

File: examples/object-initializer/object_initializer.description [hunk 2]
```diff
@@ -1 +1,4 @@
...
+// new object. There may be parameters as in any other method. However, the return type of the `__init()` method
```
Comment (MaryamZi) [inline examples/object-initializer/object_initializer.description]:
```suggestion
// new object. This method can have parameters as in any other method. However, the return type of the `__init()` method
```

File: examples/object-initializer/object_initializer.description [hunk 3]
```diff
@@ -1 +1,4 @@
...
+// should be a subtype of `error|()` which includes `()`. If an explicit definition for the `__init()` method
+// is not specified, there will be an implicit `__init()` method which does not accept any parameters and returns `()`.
```
Comment (MaryamZi) [inline examples/object-initializer/object_initializer.description]: Do we need to mention the implicit `__init()` here? Isn't it like too much info?

Comment (pubudu91) [reply examples/object-initializer/object_initializer.description]: Added it for completeness.

**Comment by pubudu91** [comment]
Issue #14969 created for the doc review of this.

**Pull request merged**